test/integration/Main.hs

```diff
@@ -610,3 +613,19 @@ testEditorFiles =
           <> fp
       )
       (removeLW t `T.isInfixOf` removeLW f)
+
+saveFileTests :: TestTree
+saveFileTests =
+  testGroup
+    "Save files"
+    [ checkLoaded "backstory" "0.6.0.0"
+    , checkLoaded "backstory" "latest"
+    ]
+ where
+  checkLoaded scenario version = testCase ("save from version " <> version) $ do
+    ef <- decodeFileEither @ScenarioInfo $ "data/test/saves/" <> scenario <> "-" <> version <> ".yaml"
+    case ef of
+      Left e -> assertFailure $ prettyPrintParseException e
+      Right si -> case si ^. scenarioStatus of
+        Played _par (Metric Completed _) _best -> pure ()
+        other -> assertFailure $ "scenario save file loaded wrong - contains: " <> show other
```

Comment:
I am open to ideas for making the test better. The regression I caused would show this failure:
```
Tests
  Save files
    save from version 0.6.0.0: FAIL
      test/integration/Main.hs:621:
      Aeson exception:
      Error in $.status.contents[1]: parsing Swarm.Game.Scenario.Scoring.GenericMetrics.Metric(Metric) failed, expected Object, but encountered Array
      Use -p '/save/&&/save from version 0.6.0.0/' to rerun this test only.
```